src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h"
+
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+
+namespace experimental {
+
+grpc_s2a_credentials_options* grpc_s2a_credentials_options_copy(
+    const grpc_s2a_credentials_options* options) {
+  if (options != nullptr && options->vtable != nullptr &&
+      options->vtable->copy != nullptr) {
+    return options->vtable->copy(options);
+  }
+  gpr_log(GPR_ERROR, "Invalid arguments to grpc_s2a_credentials_options_copy.");
+  return nullptr;
+}
+
+void grpc_s2a_credentials_options_destroy(
+    grpc_s2a_credentials_options* options) {
+  if (options != nullptr) {
+    if (options->vtable != nullptr && options->vtable->destruct != nullptr) {
```

Comment:
Now that gRPC C-core team is migrating towards C++, I am not sure if we want to keep using these C concept (e.g., v-table). F.Y.I. existing ALTS implementations need to be migrated to C++ as well.

Reply:
I've C++-ified the definition of this struct to make it look more like the `grpc_tls_credentials_options`struct that we use for SPIFFE.